sarathi/model_executor/layers/sampler.py

```diff
@@ -59,21 +62,17 @@ def forward(
 
         # Apply top-p and top-k truncation.
         top_ps, top_ks = _get_top_p_top_k(seq_metadata_list, self.vocab_size)
-        assert len(top_ps) == len(top_ks) == logits.shape[0]
-        do_top_p = any(p < 1.0 - _SAMPLING_EPS for p in top_ps)
-        do_top_k = any(k != self.vocab_size for k in top_ks)
-        if do_top_p or do_top_k:
```

Comment:
this looks like an important optimization, can we use non top-k/top-p kernel when neither top-p/top-k is requried?

Reply:
Looking further I am still inclined to say it wouldn't optimize beyond what flash infer does: https://github.com/flashinfer-ai/flashinfer/blob/46ec89317fb112dfb04bc740e711ccd3cdcf96f8/include/flashinfer/sampling.cuh#L566. But let me know if I am mistaken (that is the kernel ultimately called by the logits method too).

Reply:
But I can investigate if this code path plus https://docs.flashinfer.ai/generated/flashinfer.sampling.sampling_from_probs.html would be more efficient.

Reply:
i see, whatever seems more reasonable -- might be easier to benchmark and compare than analyzing code potentially. The PR looks pretty good to me overall, can you please share the perf numbers from benchmark.main, if everything is looking good, let's quickly merge this. cc @nitinkedia7